### PR TITLE
added header container styles to be customized

### PR DESCRIPTION
**Pull request by rghorbani**
This push adds a request to be able to center title on android ( issue #1728 ).

No specific test required to this feature.

Kept the original code formatting.

### CODE REVIEW

**Comment by brentvatne** [comment]
this is reasonable but it gets into an api design discussion. if we add this prop, what happens when someone wants to customize the `top` or `left`, etc?

I think it's reasonable to make the following changes:

`headerLayoutPreset: 'center' | 'left'` - would default to `center` on ios and `left` on android. on rtl, `left` would become `right`, in the same way `headerLeft` and `headerRight` are reversed on rtl
`headerTitleContainerStyle` - override anything on the `title` style in Header.js
`headerLeftContainerStyle`- override anything on the `left` style in Header.js
`headerRightContainerStyle` - override anything on the `right` style in Header.js

**Comment by brentvatne** [comment]
related: https://github.com/react-navigation/react-navigation/pull/4187

cc @xxsnakerxx  in case you're interested in implementing the above

**Comment by rghorbani** [comment]
@brentvatne Requested changes applied.

File: src/views/Header/Header.js [hunk 1]
```diff
@@ -211,6 +211,11 @@ class Header extends React.PureComponent {
 
     const { transitionPreset } = this.props;
 
+    let { style } = props;
+    if (options.headerLeftContainerStyle) {
+      style = [style, headerLeftContainerStyle];
```
Comment (brentvatne) [inline src/views/Header/Header.js]: should be `options.headerLeftContainerStyle` here? same for others

Comment (rghorbani) [reply src/views/Header/Header.js]: fixed.

**Comment by brentvatne** [comment]
@stantoncbradley - I'm going to finish https://github.com/react-navigation/react-navigation/pull/4588 before landing this, I'll try to get to that this weak. until then you can cherrypick this into a fork and use it if you need it!

**Comment by brentvatne** [comment]
changed my mind and landed this to keep working on related functionality in #4588

**Pull request merged**